roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -0,0 +1,105 @@
+---
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_accesskey and
+      openshift_hosted_registry_storage_s3_secretkey are required
+  when: openshift.hosted.registry.storage.s3.accesskey | default(none) is none or openshift.hosted.registry.storage.s3.secretkey | default(none) is none
+
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_bucket and
+      openshift_hosted_registry_storage_s3_region are required
+  when: openshift.hosted.registry.storage.s3.bucket | default(none) is none or openshift.hosted.registry.storage.s3.region | default(none) is none
+
+# Currently this is relying on the boto config (or env vars) on the master host, it
+# should be a local_action and use separate credentials for creating the
+# bucket (vs the ones for accessing the bucket)
```

Comment:
The above comment is only partially correct...  it is a local_action, but I think we should definitely add the ability to specify the aws creds used for this task...